sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -140,7 +140,8 @@ abstract class Optimizer(sessionCatalog: SessionCatalog, conf: CatalystConf)
   object OptimizeSubqueries extends Rule[LogicalPlan] {
     def apply(plan: LogicalPlan): LogicalPlan = plan transformAllExpressions {
       case s: SubqueryExpression =>
-        s.withNewPlan(Optimizer.this.execute(s.plan))
+        val ReturnAnswer(newPlan) = Optimizer.this.execute(ReturnAnswer(s.plan))
```

Comment:
Well extending subquery expression would make it an `Expression`, however I need a `LogicalPlan`. I will make a new node.